### PR TITLE
client: connect and publish to many accounts

### LINKED ISSUE #13: Add support for Forge twitter account
add a discord slash command and role for interacting with the forge twitter account via discord. 

this is an example of the go west slash command
Screen Shot 2022-03-25 at 4 42 19 PM

## TODO:

- [ ] Add forge twitter account API credentials (see @Soypete)
- [x] add them to existing twitter client or to a new twitter client
- [ ] update the twitter send command to distinguish between accounts
- [ ] add new slash command with the name `twitter_forge`
- [ ] add role for accessing this discord bot
- [ ] add logic give write permissions based on role

### PR DESCRIPTION
**Pull request by Soypete**

### CODE REVIEW

File: client/test_twitter_many_valid.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      "access_secret": "veNRnAWe6inFuo8o2u8SLLZLjolYDmDP7SzL0YfYI"
+    },
+    {
+	  "twitter_handle": "otherUser",
+      "consumer_key": "Dxvz1evFS4wEEPTGEFPHBog",
+      "consumer_secret": "AcSOqF21Fu85e7zjz7ZN2U4ZRhfV3WpwPAoE3Z7kBw",
+      "access_token": "370773112-GmHxMAgYyLbNEtIKZeRNFsMKPR9EyMZeS9weJAEb",
+      "access_secret": "LswwdoUaIvS8ltyTt5jkRh4J50vUPVVHtR2YPi5kE"
+    }
+  ]
```
Comment (Soypete) [inline client/test_twitter_many_valid.json:17]: credential from twitter api
https://developer.twitter.com/en/docs/authentication/oauth-1-0a/creating-a-signature

File: client/setup.go [hunk 2]
```diff
@@ -11,23 +13,19 @@ import (
...
-	config := oauth1.NewConfig(oauthConsumerKey, oauthConsumerSecret)
-	token := oauth1.NewToken(oauthAccessToken, oauthAccessSecret)
 	discordToken := os.Getenv("DISCORD_TOKEN")
-	httpClient := config.Client(oauth1.NoContext, token)
 
-	// Twitter client
-	client := twitter.NewClient(httpClient)
+	tClients, err := SetupTwitterClients(os.Getenv("CREDENTIAL_FILE"))
+	if err != nil {
+		log.Println("credential file support not implemented")
```
Comment (jnewmano) [inline client/setup.go]: print the error message out too?

Comment (jnewmano) [reply client/setup.go]: Return the error, instead of logging?

File: client/setup.go [hunk 3]
```diff
@@ -11,23 +13,19 @@ import (
...
+	}
 
 	//Discord client
 	dgclient, err := setupDiscord(discordToken)
```
Comment (jnewmano) [inline client/setup.go:29]: missing an error check here

File: client/setup.go [hunk 4]
```diff
@@ -37,10 +35,44 @@ func NewClient() (*Client, error) {
...
+		ApiCredentials []struct {
+			TwitterHandle       string `json:"twitter_handle"`
+			OauthConsumerKey    string `json:"consumer_key"`
+			OauthConsumerSecret string `json:"consumer_secret"`
+			OauthAccessToken    string `json:"access_token"`
+			OauthAccessSecret   string `json:"access_secret"`
+		} `json:"twitter_clients"`
+	}
+
+	f, err := os.Open(jsonFileName)
```
Comment (jnewmano) [inline client/setup.go:58]: not super important, but it's a good habit to make sure you close file handles when you open them

File: internal/botguts/youtubevids.go [hunk 5]
```diff
@@ -46,29 +49,49 @@ func (ab *AutoBot) Close() {
...
+			hashtags = "#k8s #kubernetes #cncf #cloudnative"
+		case "Utah Go User Group":
+			hashtags = "#golang #meetup"
+		case "Mlops Utah Meetup":
+			hashtags = "#machinelearning #mlops #ai #data #ml"
+		case "Women Who Go Utah Meetup":
+			hashtags = "#golang #wwg @wwg #womenintech"
+		case "Front End Craftsmen":
+			hashtags = "#Frontend #javascript #js #react"
+		default:
```
Comment (jnewmano) [inline internal/botguts/youtubevids.go]: handle the default case by returning an error?

**Pull request merged**